### PR TITLE
Write field-less union case name without formatting

### LINKED ISSUE #58: Preserve formatting when serializing discriminated unions without fields
When serializing discriminated unions without any fields, the current behaviour is to use the Newtonsoft.Json `ContractResolver`'s `GetResolvedPropertyName` for serialization of the case name.

This means that when serializing using `CamelCasePropertyNamesContractResolver`
```
type FooBar = Foo | Bar

{
    Foo: Bar
}
```
Becomes:
```
{
    "foo": "bar"
}
```

I would prefer that the serialization instead produces this:
```
{
    "foo": "Bar"
}
```

As I believe it better conveys the value.

The code causing this is likely at or near [Compact.fs:151](https://github.com/Microsoft/fsharplu/blob/master/FSharpLu.Json/Compact.fs#L151).

### PR DESCRIPTION
**Pull request by erik-lundgren**
Resolves #58 

Sorry about opening two PRs, got the wrong author in there accidentally.

### CODE REVIEW

File: FSharpLu.Tests/JsonTests.fs [hunk 1]
```diff
@@ -343,14 +343,14 @@ type JsonSerializerTests() =
     member __.``CamelCaseSerializer handles discriminated unions with consecutive uppercase characters`` () =
         let du = [ FOO ; FOOWithRecord { BAR=2; BAZNumber=3 } ]
         let str = CamelCaseSerializer.serialize du
-        Assert.AreEqual("""["foo",{"fooWithRecord":{"bar":2,"bazNumber":3}}]""", str)
+        Assert.AreEqual("""["FOO",{"fooWithRecord":{"bar":2,"bazNumber":3}}]""", str)
```
[the remark this answers is not in the document]
Comment (blumu) [reply FSharpLu.Tests/JsonTests.fs]: Thanks for clarifying in #58  Closing this comment

File: FSharpLu.Json/Compact.fs [hunk 2]
```diff
@@ -148,7 +148,7 @@ type CompactUnionJsonConverter(?tupleAsHeterogeneousArray:bool) =
 
             match fields with
             // Field-less union case
-            | [||] -> writer.WriteValue(convertName case.Name)
+            | [||] -> writer.WriteValue(case.Name)
```
[the remark this answers is not in the document]
Comment (blumu) [reply FSharpLu.Json/Compact.fs]: I would recommend adding an optional parameter to the type `CompactUnionJsonConverter` to indicate the casing behaviour for values:
For example:
```fs
type CompactUnionJsonConverter(?tupleAsHeterogeneousArray:bool, ?camelCaseValues:bool) =
      let tupleAsHeterogeneousArray:bool= defaultArg tupleAsHeterogeneousArray false
      ...
     | [||] when tupleAsHeterogeneousArray  -> writer.WriteValue(case.Name)
     | [||] when not tupleAsHeterogeneousArray  -> writer.WriteValue(convertName case.Name)
      ...
```

File: FSharpLu.Tests/JsonTests.fs [hunk 3]
```diff
@@ -343,14 +343,14 @@ type JsonSerializerTests() =
...
 
     [<TestMethod>]
     [<TestCategory("FSharpLu.Json.CamelCase")>]
     member __.``CamelCaseSerializer handles option type`` () =
         let du = { WrappedField = Some Red }
         let str = CamelCaseSerializer.serialize du
-        Assert.AreEqual("""{"wrappedField":"red"}""", str)
+        Assert.AreEqual("""{"wrappedField":"Red"}""", str)
 
     [<TestMethod>]
```
Comment (blumu) [inline FSharpLu.Tests/JsonTests.fs]: Could you add a separate unit test to cover the expected casing behaviour for a DU with both field and fieldless cases?

Comment (erik-lundgren) [reply FSharpLu.Tests/JsonTests.fs]: Added such a test here: https://github.com/microsoft/fsharplu/pull/60/files#diff-3b07ffc2b15856686373ed5b6a3b98c6R351

File: FSharpLu.Json/Compact.fs (hunk 2 above)
[the remark this answers is not in the document]
Comment (erik-lundgren) [reply FSharpLu.Json/Compact.fs]: I added the optional parameter, but renamed it to `usePropertyFormatterForValues` which reversed its meaning so for non-breaking purposes it is now default `true`. It describes more clearly what is happening, I think, as the property formatter is not guaranteed to use camelCase when formatting. I hope this still stays in line with your suggestion.

**Pull request merged**